networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -100,15 +103,17 @@ def get_hostgroup_mode(self, context, hostgroup_name):
         hg_modes = self.get_hostgroup_modes(context, [hostgroup_name])
         return hg_modes.get(hostgroup_name)
 
+    @db_api.CONTEXT_READER
     def set_hostgroup_mode(self, context, hostgroup_name, hostgroup_mode):
-        with context.session.begin(subtransactions=True):
+        with db_api.CONTEXT_WRITER.using(context):
             query = context.session.query(HostgroupModeModel).filter(HostgroupModeModel.hostgroup == hostgroup_name)
             hg = query.first()
             if not hg:
                 return False
             hg.mode = hostgroup_mode
         return True
```

Comment:
This has to be a writer. I wonder why we don't do an explicit add of the updated object. Does that still work with the new `db_api`?